src/Control/Map_Edit.cpp

```diff
@@ -122,12 +128,10 @@ bool CMapView::HandleOnMouseMoveShapeEditor(int x, int y, long nFlags)
 	}
 	else 
 	{
-		tkEditorBehavior behavior;
-		_shapeEditor->get_EditorBehavior(&behavior);
+		tkSnapMode snapMode;
+		_shapeEditor->get_SnapMode(&snapMode);
 		EditorBase* base = GetEditorBase();
 		bool handled = false;
-		double projX, projY;
-		this->PixelToProjection(x, y, projX, projY);
 
 		// highlighting of vertices
 		if (behavior == ebVertexEditor)

```

Comment:
I have to look further into this - 'behavior' was originally defined (within this block of code) as tkEditorBehavior, but you have redefined it (above) as tkLayerSelection, using get_SnapBehavior.  It seems that this test may not be giving back the expected result.